src/IRTS/CodegenC.hs

```diff
@@ -179,8 +179,10 @@ bcc i (ASSIGNCONST l c)
     = indent i ++ creg l ++ " = " ++ mkConst c ++ ";\n"
   where
     mkConst (I i) = "MKINT(" ++ show i ++ ")"
-    mkConst (BI i) | i > ((-2)^30) && i < (2^30) = "MKINT(" ++ show i ++ ")"
-                   | otherwise = "MKBIGC(vm,\"" ++ show i ++ "\")"
+    mkConst (BI i) = let maxInt = 2^30
```

Comment:
On C at least the maximum integer is `INT_MAX` which is equal to 2^31-1.
The minimum integer is `INT_MIN` which is equal to -2^31.

see https://www.tutorialspoint.com/c_standard_library/limits_h.htm

Reply:
Well for 32-bit integers anyway.

Reply:
That's true, but the last bit of an int is used to tag it as an unboxed value in the runtime.

Reply:
Ah, I see.

Reply:
Well, since the size is based on a pointer, it’s either 31 or 63 bits depending on platform.